clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystemUtils.java

```diff
@@ -134,4 +142,42 @@ public static boolean waitCompleted(final TachyonFileSystemCore tfs, final Tachy
 
     return completed;
   }
+
+  /**
+   * Persist the given file to the under file system.
+   *
+   * @param tfs {@link TachyonFileSystem} to carry out tachyon operations
+   * @param file the file to persist
+   * @param fileInfo the file info of the file
+   * @param tachyonConf TachyonConf object
+   * @return the size of the file persisted
+   * @throws IOException if an I/O error occurs
+   * @throws FileDoesNotExistException if the given file does not exist
+   * @throws TachyonException if an unexpected Tachyon error occurs
+   */
+  public static long persistFile(TachyonFileSystem tfs, TachyonFile file, FileInfo fileInfo,
+      TachyonConf tachyonConf) throws IOException, FileDoesNotExistException, TachyonException {
+    Closer closer = Closer.create();
+    long ret;
+    try {
+      InStreamOptions inStreamOptions = new InStreamOptions.Builder(tachyonConf)
+          .setTachyonStorageType(TachyonStorageType.NO_STORE).build();
+      FileInStream in = closer.register(tfs.getInStream(file, inStreamOptions));
+      TachyonURI dstPath = new TachyonURI(fileInfo.getUfsPath());
+      UnderFileSystem ufs = UnderFileSystem.get(dstPath.getPath(), tachyonConf);
+      String parentPath = dstPath.getParent().getPath();
+      if (!ufs.exists(parentPath) && !ufs.mkdirs(parentPath, true)) {
```

Comment:
If we use the setState RPC, we could let the server-side to persist the file, which is consistent with other code paths that persist objects created through Tachyon.